pulsar-functions/instance/src/main/python/python_instance_main.py

```diff
@@ -88,8 +132,10 @@ def main():
   parser.add_argument('--extra_dependency_repository', required=False, help='For packaged python like wheel files, any extra repository to pull the dependencies from')
   parser.add_argument('--state_storage_serviceurl', required=False, help='Managed State Storage Service Url')
   parser.add_argument('--cluster_name', required=True, help='The name of the cluster this instance is running on')
```

Comment:
Should the arugments be declared required ? If a config is provided in file, then maybe the config from command line is not required any more

Reply:
you're right, I adjust all args to `required=False` and add a new function `validate_arguments()` to verify the merged args, mainly to check the mandatory args.